doc/README.txt

```diff
@@ -0,0 +1,20 @@
+Adding an interface
+-------------------
+
+Adding a plugin
+---------------
+
+Adding a quantum operation
+--------------------------
+
+Adding a quantum measurement
+----------------------------
+
+Adding a template
+-----------------
+
+Adding an optimizer
+-------------------
+
+Adding a module
+---------------
```

Comment:
@mariaschuld, I'm assuming this file can be deleted?

Reply:
I renamed and added the instructions on where to change the docs when adding new functionality. Feel free to delete if you find it obsolete.